### PR TITLE
add 0xCB New Horizons - with rp2040 and 32u4 support

### PR DESCRIPTION
**Pull request by Conor-Burns**
## Description
This PR adds support for the 0xCB New Horizons:

https://keeb.supply/products/0xcb-new-horizons

## Types of Changes

- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/0xcb/newhorizons/info.json
Comment (waffle87) [inline keyboards/0xcb/newhorizons/info.json:1]: Please correct the `x`/`y` coordinates for the following layouts. (the keys all in one line is not valid)
```
LAYOUT_bigspace
LAYOUT_bigspace_blocker
LAYOUT_bigspace_blocker_iso
LAYOUT_bigspace_iso
LAYOUT_hugespace
LAYOUT_hugespace_iso
LAYOUT_split
LAYOUT_split_blocker
LAYOUT_split_blocker_iso
LAYOUT_split_iso
```

File: keyboards/0xcb/newhorizons/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        KC_LCTL, KC_LGUI, KC_LALT,                   KC_SPC,  KC_SPC,  KC_SPC,                    MO(1),   KC_RALT, KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+    [1] = LAYOUT_all(
+        _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, _______, _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
+        _______, _______, _______,                   _______, _______, _______,                   _______, _______, _______, _______, _______, _______
+    ),
+    [2] = LAYOUT_all(
```
Comment (waffle87) [inline keyboards/0xcb/newhorizons/keymaps/default/keymap.c:22]: This layer is not accessible.

File: keyboards/0xcb/newhorizons/readme.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+65% Keyboard
+
+* Keyboard Maintainer: [Conor Burns](https://github.com/conor-burns)
+* Hardware Supported: https://github.com/0xcb-dev/0xCB-NewHorizons
+* Hardware Availability: On [KeebSupply](https://keeb.supply/products/0xcb-new-horizons)
+
+![](https://keeb.supply/assets/nh-full.webp)
+
+![](https://keeb.supply/assets/nh-bottom.webp)
```
Comment (waffle87) [inline keyboards/0xcb/newhorizons/readme.md:11]: These images need to be rehosted on Imgur.

File: keyboards/0xcb/newhorizons/32u4/info.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{
+    "bootloader": "qmk-dfu",
+    "ws2812": {
+        "pin": "F1"
+    },
+    "encoder": {
+        "enabled": true,
+        "rotary": [
+            {"pin_a": "D5", "pin_b": "C7", "resolution": 4}
```
Comment (waffle87) [inline keyboards/0xcb/newhorizons/32u4/info.json:9]:
```suggestion
        "rotary": [
            {"pin_a": "D5", "pin_b": "C7"}
```

File: keyboards/0xcb/newhorizons/rp2040_ce/info.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
+{
+    "processor": "RP2040",
+    "bootloader": "rp2040",
+    "encoder": {
+        "enabled": true,
+        "rotary": [
+            {
+                "pin_a": "GP13",
+                "pin_b": "GP14",
+                "resolution": 4
```
Comment (waffle87) [inline keyboards/0xcb/newhorizons/rp2040_ce/info.json:10]:
```suggestion
        "rotary": [
            {
                "pin_a": "GP13",
                "pin_b": "GP14"
```

File: keyboards/0xcb/newhorizons/rp2040_ce/info.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        "console": false,
+        "encoder": true,
+        "encoder_map": true,
+        "extrakey": true,
+        "key_override": false,
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true,
+        "sleep_led": false,
+        "tap_dance": false
```
Comment (waffle87) [inline keyboards/0xcb/newhorizons/rp2040_ce/info.json:29]:
```suggestion
        "audio": false,
        "bootmagic": true,
        "encoder": true,
        "extrakey": true,
        "mousekey": true,
        "nkro": true,
        "rgb_matrix": true,
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/0xcb/newhorizons/info.json [hunk 6]
```diff
@@ -0,0 +1,870 @@
...
+            {"x": 164, "y": 19, "flags": 2},
+            {"x": 238, "y": 19, "flags": 2},
+            {"x": 0, "y": 38, "flags": 2},
+            {"x": 224, "y": 38, "flags": 2},
+            {"x": 45, "y": 58, "flags": 2},
+            {"x": 67, "y": 58, "flags": 2},
+            {"x": 164, "y": 58, "flags": 2},
+            {"x": 238, "y": 58, "flags": 2}
+        ],
+        "max_brightness": 255
```
[the remark this answers is not in the document]
Comment (zvecr) [reply keyboards/0xcb/newhorizons/info.json:30]: `led_count` isnt supported within `rgb_matrix`, however it is generated from length of of the `rgb_matrix.layout` array.

File: keyboards/0xcb/newhorizons/info.json [hunk 7]
```diff
@@ -0,0 +1,870 @@
...
+                {"label": "2,3", "matrix": [2, 3], "x": 6.5, "y": 1},
+                {"label": "3,3", "matrix": [3, 3], "x": 7.5, "y": 1},
+                {"label": "2,4", "matrix": [2, 4], "x": 8.5, "y": 1},
+                {"label": "3,4", "matrix": [3, 4], "x": 9.5, "y": 1},
+                {"label": "2,5", "matrix": [2, 5], "x": 10.5, "y": 1},
+                {"label": "3,5", "matrix": [3, 5], "x": 11.5, "y": 1},
+                {"label": "2,6", "matrix": [2, 6], "x": 12.5, "y": 1},
+                {"label": "3,6", "matrix": [3, 6], "x": 13.5, "y": 1, "w": 1.5},
+                {"label": "3,7", "matrix": [3, 7], "x": 15, "y": 1},
+                {"label": "4,0", "matrix": [4, 0], "x": 0, "y": 2, "w": 1.25},
```
Comment (dunk2k) [inline keyboards/0xcb/newhorizons/info.json:66]:
```suggestion
                {"label": "4,0", "matrix": [4, 0], "x": 0, "y": 2, "w": 1.75},
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/0xcb/newhorizons/config.h [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+#define ENABLE_RGB_MATRIX_BAND_VAL
+#define ENABLE_RGB_MATRIX_BAND_SPIRAL_VAL
+#define ENABLE_RGB_MATRIX_CYCLE_ALL
+#define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
+#define ENABLE_RGB_MATRIX_CYCLE_UP_DOWN
+#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN
+#define ENABLE_RGB_MATRIX_CYCLE_PINWHEEL
+#define ENABLE_RGB_MATRIX_RAINBOW_PINWHEELS
+#define ENABLE_RGB_MATRIX_RAINDROPS
+
```
Comment (dunk2k) [inline keyboards/0xcb/newhorizons/config.h:25]: migrate to `info.json`
```suggestion
```

File: keyboards/0xcb/newhorizons/info.json [hunk 9]
```diff
@@ -0,0 +1,870 @@
...
+        "pid": "0x2006",
+        "vid": "0xCB00"
+    },
+    "debounce": 5,
+    "diode_direction": "ROW2COL",
+    "qmk": {
+        "tap_keycode_delay": 10
+    },
+    "rgb_matrix": {
+        "driver": "ws2812",
```
Comment (dunk2k) [inline keyboards/0xcb/newhorizons/info.json:16]:
```suggestion
        "driver": "ws2812",
        "animations": {
            "alphas_mods": true,
            "gradient_up_down": true,
            "gradient_left_right": true,
            "breathing": true,
            "band_val": true,
            "band_spiral_val": true,
            "cycle_all": true,
            "cycle_left_right": true,
            "cycle_up_down": true,
            "cycle_out_in": true,
            "cycle_pinwheel": true,
            "rainbow_pinwheels": true,
            "raindrops": true
        },
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Pull request closed without merging**